applications/minotari_ledger_wallet/wallet/src/utils.rs

```diff
@@ -147,32 +144,32 @@ fn cx_error_to_string(e: CxError) -> String {
 }
 
 // Get a raw 64 byte key hash from the BIP32 path.
-// - The wrapper function for the syscall `os_perso_derive_node_bip32`, `bip32_derive`, requires a 96 byte buffer when
-//   called with `CurvesId::Ed25519` as it checks the consistency of the curve choice and key length in order to prevent
-//   the underlying syscall from panicking.
-// - The syscall `os_perso_derive_node_bip32` returns 96 bytes as:
-//     private key: 64 bytes
-//     chain: 32 bytes
-//   Example:
-//     d8a57c1be0c52e9643485e77aac56d72fa6c4eb831466c2abd2d320c82d3d14929811c598c13d431bad433e037dbd97265492cea42bc2e3aad15440210a20a2d0000000000000000000000000000000000000000000000000000000000000000
-//  - This function applies domain separated hashing to the 64 byte private key of the returned buffer to get 64
-//    uniformly distributed random bytes.
-fn get_raw_key_hash(path: &[u32]) -> Result<Zeroizing<[u8; 64]>, String> {
-    let mut key_buffer = Zeroizing::new([0u8; 96]);
-    const BIP32_KEY_LENGTH: usize = 64;
-    let raw_key_64 = match bip32_derive(CurvesId::Ed25519, path, key_buffer.as_mut(), Some(&mut [])) {
+fn get_raw_bip32_key(path: &[u32], curve: CurvesId) -> Result<Zeroizing<[u8; 64]>, String> {
+    match curve {
+        CurvesId::Secp256k1 | CurvesId::Ed25519 => {},
+        _ => return Err("Unsupported curve".to_string()),
+    }
+    let mut key_buffer = Zeroizing::new([0u8; 64]);
+
+    match bip32_derive(curve, path, key_buffer.as_mut(), Some(&mut [])) {
         Ok(_) => {
-            let binding = &key_buffer.as_ref()[..BIP32_KEY_LENGTH];
-            if binding == &[0u8; BIP32_KEY_LENGTH] {
+            let binding = &key_buffer.as_ref();
+            if binding == &[0u8; 64] {
                 return Err(cx_error_to_string(CxError::InternalError));
             }
-            let mut key_bytes = Zeroizing::new([0u8; BIP32_KEY_LENGTH]);
+            let mut key_bytes = Zeroizing::new([0u8; 64]);
             // `copy_from_slice` will not panic as the length of the slice is equal to the length of the array
             key_bytes.as_mut().copy_from_slice(binding);
-            key_bytes
+            Ok(key_bytes)
```

Comment:
This seems unnecessary. Why not directly test `key_buffer` against the zero slice and then return it directly?

Reply:
Good catch, `key_buffer ` is already of the correct type.